frappe/core/doctype/recorder_request/recorder_request.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2023, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+import json
+import re
+from collections import Counter
+
+import sqlparse
+
+import frappe
+from frappe import _
+from frappe.database.database import is_query_type
+from frappe.model.document import Document
+
+RECORDER_INTERCEPT_FLAG = "recorder-intercept"
+RECORDER_REQUEST_SPARSE_HASH = "recorder-requests-sparse"
+RECORDER_REQUEST_HASH = "recorder-requests"
+TRACEBACK_PATH_PATTERN = re.compile(".*/apps/")
```

Comment:
We can leave `recorder.py` code as is and just import what is necessary IMO. Backend code won't require any changes for this PR.